src/reducers/computeRenderedColumns.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright Schrodinger, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule computeRenderedColumns
+ */
+
+'use strict';
+
+import clamp from 'lodash/clamp';
+import columnWidths from 'columnWidths';
+import scrollbarsVisibleSelector from 'scrollbarsVisible';
+import { updateColumnWidth, updateColumnGroupWidth } from 'updateColumnWidth';
+
+export default function computeRenderedColumns(state, columnAnchor) {
+  // clone state
+  const newState = Object.assign({}, state);
+  
+  // get buffer and viewport range
+  const columnRange = calculateRenderedColumnRange(newState, columnAnchor);
+  
+  // update the offsets and buffer mapping
+  computeRenderedColumnOffsets(newState, columnRange, newState.scrolling);
+
+  // scrollX might have changed due to change in columns and offsets
+  let scrollX = state.scrollX;
+  if (columnRange.firstViewportCol !== columnRange.endViewportCol) {
+    scrollX = newState.columnOffsets[columnRange.firstViewportCol] - newState.firstColumnOffset;
+  }
+  const { maxScrollX } = columnWidths(state);
+  newState.scrollX = clamp(scrollX, 0, maxScrollX);
+
+  return newState;
+}
+
+/**
+ * Determine the range of columns to render (buffer and viewport)
+ * The leading and trailing buffer is based on a fixed count,
+ * while the viewport columns are based on their width and the viewport width.
+ * We use the columnAnchor to determine what either the first or last column
+ * will be, as well as the offset.
+ *
+ * NOTE (jordan) This alters state so it shouldn't be called
+ * without state having been cloned first.
+ *
+ * @param {!Object} state
+ * @param {{
+ *   firstIndex: number,
+ *   firstOffset: number,
+ *   lastIndex: number,
+ * }} columnAnchor
+ * @return {{
+ *   endBufferCol: number,
+ *   endViewportCol: number,
+ *   firstBufferCol: number,
+ *   firstViewportCol: number,
+ * }}
+ * @private
+ */
+function calculateRenderedColumnRange(state, columnAnchor) {
```

Comment:
Similar to [calculateRenderedRowRange](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/reducers/computeRenderedRows.js#L105)

**Quick description:**
Since our columnAnchor already has the first/last column, we'll simply fill our table with the columns one by one.

So if first column is at 4, and table size is 1000, we'll add 4th column, then 5th column, and so on, until the widths sum upto (or greater than) 1000. This will give the range of columns.